src/status_im/contexts/settings/wallet/data_store.cljs

```diff
@@ -0,0 +1,47 @@
+(ns status-im.contexts.settings.wallet.data-store)
+
+(defn extract-keypair-name
+  [db key-uids-set]
+  (when (= (count key-uids-set) 1)
+    (let [key-uid  (first key-uids-set)
+          keypairs (get-in db [:wallet :keypairs])]
+      (->> (filter #(= (:key-uid %) key-uid) keypairs)
+           first
+           :name))))
+
+(defn update-keypair
+  [keypairs key-uid update-fn]
+  (mapcat (fn [keypair]
+            (if (= (keypair :key-uid) key-uid)
+              (if-let [updated (update-fn keypair)]
+                [updated]
+                [])
+              [keypair]))
+   keypairs))
```

Comment:
I know this is super annoying 😢 . I will make a PR to re-structure the keypairs in app-db once https://github.com/status-im/status-mobile/pull/20160 is merged.